pypeit/pypeit.py

```diff
@@ -811,20 +811,32 @@ def objfind_one(self, frames, det, bg_frames=None, std_outfile=None):
             sciImg = bkg_redux_sciimg.sub(bgimg)
 
         # Flexure
-        spat_flexure = None
+        spat_flexure = np.zeros((self.caliBrate.slits.nslits, 2))  # No spatial flexure, unless we find it below
         # use the flexure correction in the "shift" column
         manual_flexure = self.fitstbl[frames[0]]['shift']
-        if (self.objtype == 'science' and self.par['scienceframe']['process']['spat_flexure_correct']) or \
-                (self.objtype == 'standard' and self.par['calibrations']['standardframe']['process']['spat_flexure_correct']) or \
+        if (self.objtype == 'science' and self.par['scienceframe']['process']['spat_flexure_correct'] != "none") or \
+                (self.objtype == 'standard' and self.par['calibrations']['standardframe']['process']['spat_flexure_correct'] != "none") or \
                     manual_flexure:
-            if (manual_flexure or manual_flexure == 0) and not (np.issubdtype(self.fitstbl[frames[0]]["shift"], np.integer)):
-                msgs.info(f'Implementing manual flexure of {manual_flexure}')
-                spat_flexure = np.float64(manual_flexure)
+            if (manual_flexure != self.fitstbl.MASKED_VALUE) and np.issubdtype(self.fitstbl[frames[0]]["shift"], np.integer):
```

Comment:
a `not` is missing. It should be:
`if (manual_flexure != self.fitstbl.MASKED_VALUE) and not np.issubdtype(self.fitstbl[frames[0]]["shift"], np.integer):`

Reply:
I'm not quite sure about this. My understanding of the previous code is:

`(manual_flexure or manual_flexure == 0)`
if `manual_flexure` is anything other than `None`

`and not (np.issubdtype(self.fitstbl[frames[0]]["shift"], np.integer))`
and it's set to something other than an integer.

The first line seems like this is always true, since the previous default in `metadata.py` would set `shift=0` as the default (unset) value. I think the `MASKED_VALUE` clears this up. Then, the second line has me more confused why this is there... For example, the simple examples:
```
In [2]: not np.issubdtype(np.zeros(2, dtype=float).dtype, np.integer)
Out[2]: True

In [3]: not np.issubdtype(np.zeros(2, dtype=int).dtype, np.integer)
Out[3]: False
```
So, I think this `if` statement is only being triggered when the `shift` value is a float. I don't really understand why this is, but I assumed folks would input a integer pixel value (hence my change). Perhaps this should just be:

`if manual_flexure != self.fitstbl.MASKED_VALUE:`

Reply:
Ok. I'm confused too. 
I think you are right, just checking that `self.fitstbl[frames[0]]['shift']` is not equal to `self.fitstbl.MASKED_VALUE` (or possibly `None`) should suffice.

BTW, it just occurred to me that, currently, the manual flexure can be only applied to the science/standard frame, since this piece of code is here and not in `rawimage.py`. This is a bit misleading because a `shift` column is added to the `.pypeit` file and the users could think that they can set the shift also for other frames. Sorry, this has nothing to do with your PR, but I think it's something that we need to fix at certain point.